library/src/main/java/com/android/gpstest/library/data/SharedSensorManager.kt

```diff
@@ -133,7 +133,7 @@ class SharedSensorManager constructor(
         try {
             if (SatelliteUtils.isRotationVectorSensorSupported(context)) {
                 // Use the modern rotation vector sensors
-                val vectorSensor: Sensor =
+                val vectorSensor: Sensor? =
```

Comment:
Would you prefer to check on line 137 with `sensorManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR)!!` ?